File: src/main/java/com/github/eendroroy/sdk/openweather/interceptor/OWInterceptor.java
package com.github.eendroroy.sdk.openweather.interceptor;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * @author indrajit
 */

public interface OWInterceptor extends Interceptor {
    String rawRequestBody(Request request);
    String rawResponseBody(Response response);
}

File: src/main/kotlin/com/github/eendroroy/sdk/openweathermap/interceptor/OWInterceptor.kt
package com.github.eendroroy.sdk.openweathermap.interceptor

import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response

/**
 * @author indrajit
 */
interface OWInterceptor : Interceptor {
    fun rawRequestBody(request: Request): String
    fun rawResponseBody(response: Response): String
}